### ISSUE TITLE
WinError 267] The directory name is invalid

### ISSUE THREAD

**Issue by Mohamed-Elmasri** [bython17/cwm-downloader#2]
I follow every step 
but I got this error..

(cwm-downloader-py3.11) PS C:\Users\mohamed.elmasri\cwm-downloader\cwm\cwm-downloader> cwm-downloader download https://codewithmosh.com/courses/639884/lectures/11425294
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ C:\Users\mohamed.elmasri\cwm-downloader\cwm\cwm-downloader\cwm_downloader\cli.py:114 in download │
│                                                                                                  │
│   111 │   │   # by setting the headers and cookies from the credentials.json file for us.        │
│   112 │   │   with initialize_session() as session:                                              │
│   113 │   │   │   course_obj = Course(url, session, timeout)                                     │
│ ❱ 114 │   │   │   _download(course_obj, section_no, lecture_no, only, path, chunk_size=chunk_s   │
│   115 │   # This is an error raised by the url validator found in the base abstract class        │
│   116 │   # Scraper in _scraper.py                                                               │
│   117 │   except IncorrectUrlError:                                                              │
│                                                                                                  │
│ C:\Users\mohamed.elmasri\cwm-downloader\cwm\cwm-downloader\cwm_downloader\cli.py:80 in _download │
│                                                                                                  │
│    77 │   │   if not lecture_no:                                                                 │
│    78 │   │   │   lecture_no = 1                                                                 │
│    79 │   │                                                                                      │
│ ❱  80 │   │   course_obj.download(base_dir, section_no-1, lecture_no-1, **download_args)         │
│    81                                                                                            │
│    82                                                                                            │
│    83 @app.callback()                                                                            │
│                                                                                                  │
│ C:\Users\mohamed.elmasri\cwm-downloader\cwm\cwm-downloader\cwm_downloader\utils.py:161 in        │
│ decorated_func                                                                                   │
│                                                                                                  │
│   158 │   @wraps(func)                                                                           │
│   159 │   def decorated_func(*args, **kwargs):                                                   │
│   160 │   │   try:                                                                               │
│ ❱ 161 │   │   │   return func(*args, **kwargs)                                                   │
│   162 │   │   except IndexError:                                                                 │
│   163 │   │   │   render_message('error', 'The specified section or lecture doesn\'t exist.')    │
│   164 │   │   │   raise typer.Exit(1)                                                            │
│                                                                                                  │
│ C:\Users\mohamed.elmasri\cwm-downloader\cwm\cwm-downloader\cwm_downloader\scraper\course_scraper │
│ .py:98 in download                                                                               │
│                                                                                                  │
│    95 │   │   │                                                                                  │
│    96 │   │   │   # Create the course directory inside the base_dir provided.                    │
│    97 │   │   │   course_dir = base_dir / str(self)                                              │
│ ❱  98 │   │   │   course_dir.mkdir(exist_ok=True)                                                │
│    99 │   │   │                                                                                  │
│   100 │   │   │   # Loop over the enumerations of the selected sections to get the indexes and   │
│   101 │   │   │   # downloads an indivisual section using the seciton number.                    │
│                                                                                                  │
│ C:\Users\mohamed.elmasri\AppData\Local\Programs\Python\Python311\Lib\pathlib.py:1116 in mkdir    │
│                                                                                                  │
│   1113 │   │   Create a new directory at this given path.                                        │
│   1114 │   │   """                                                                               │
│   1115 │   │   try:                                                                              │
│ ❱ 1116 │   │   │   os.mkdir(self, mode)                                                          │
│   1117 │   │   except FileNotFoundError:                                                         │
│   1118 │   │   │   if not parents or self.parent == self:                                        │
│   1119 │   │   │   │   raise                                                                     │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
NotADirectoryError: [WinError 267] The directory name is invalid: 'The Ultimate Data Structures & Algorithms: Part 1'
(cwm-downloader-py3.11) PS C:\Users\mohamed.elmasri\cwm-downloader\cwm\cwm-downloader>

**Comment by Mohamed-Elmasri** [comment]
I try to use powershell and I got this message

(cwm-downloader-py3.11) PS C:\Users\mohamed.elmasri\cwm-downloader\test2\cwm-downloader> cwm-downloader download https://codewithmosh.com/courses/639884/lectures/11425294
Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.
ERROR The program could'nt fetch resources. There might be updates to the site or your subscription has ended. Or you
might have an invalid cookies
(cwm-downloader-py3.11) PS C:\Users\mohamed.elmasri\cwm-downloader\test2\cwm-downloader>

and I am sure that everystep is ok
